### ISSUE TITLE
Scopes are encoded incorrectly when passed as an array

### ISSUE THREAD

**Issue by cjlarose** [lelylan/simple-oauth2#186]
The README suggests that for the client credentials access token request and the resource owner  credentials access token request, multiple scopes can be specified as an array. However, doing so causes the scopes to be encoded incorrectly in the request body.

For example, using the client credentials grant with the `application/x-www-form-urlencoded` request content-type:

```javascript
const credentials = {
  client: { id: 'client-id', secret: 'client-secret' },
  auth: { tokenHost: 'http://localhost:3000' },
};
const oauth2 = require('simple-oauth2').create(credentials);
oauth2.clientCredentials.getToken({ scope: ['scope-1', 'scope-2'] });
```

Produces a request with the body

```
scope=scope-1&scope=scope-2&grant_type=client_credentials
```

I'd instead expect the request body to contain the scopes as space-separated like so:

```
scope=scope-1%20scope-2&grant_type=client_credentials
```

...to be consistent with [RFC 6749 § 3.3][spec]

[spec]: https://tools.ietf.org/html/rfc6749#section-3.3

**Comment by qualbeen** [comment]
This seems to still be buggy; the code does indeed still join the array with a comma. 😠 
Current code from master: 
```
    if (Array.isArray(params.scope)) {
      params.scope = params.scope.join(',');
    }
```
source: [`/master/lib/client/auth-code.js`](https://github.com/lelylan/simple-oauth2/blob/master/lib/client/auth-code.js#L30-L32)

A previous commit from @hugopeixoto resolved this (commit https://github.com/lelylan/simple-oauth2/pull/200/commits/28df029877c6e829e8f092d22cd206c1a6fa66a8), but where did the commit go? Not merged?

**Comment by jonathansamines** [comment]
@qualbeen Since this is a breaking change, it is still planned for the next major release. Will hopefully will be out in a couple of weeks.

**Comment by LinusU** [comment]
@jonathansamines would be very nice to get this out, this is currently making the library incompatible with Google auth when using more than one scope 😞 

(thanks for an otherwise great lib btw. 🍻)

**Comment by cjlarose** [comment]
Just in case it's not clear to someone browsing this issue, it's still possible as a user of this library to pass in your scopes as a space-separated string yourself.

```javascript
const credentials = {
  client: { id: 'client-id', secret: 'client-secret' },
  auth: { tokenHost: 'http://localhost:3000' },
};
const oauth2 = require('simple-oauth2').create(credentials);
oauth2.clientCredentials.getToken({ scope: 'scope-1 scope-2'] });
```

will correctly produce 

```
scope=scope-1%20scope-2&grant_type=client_credentials
```

in the request body.

This way if you're talking to an OAuth server that implements the specification exactly, you can still request multiple scopes.

**Comment by dawadam** [comment]
It might be useful to add parametter to configure the "scope join" character.

**Comment by jonathansamines** [comment]
Solved by #256
